crates/ide-completion/src/render.rs

```diff
@@ -334,21 +336,45 @@ fn render_resolution_path(
             }
         }
     }
-    if let ScopeDef::Local(local) = resolution {
-        let ty = local.ty(db);
+
+    let mut set_item_relevance = |is_local: bool, ty: Type| {
         if !ty.is_unknown() {
             item.detail(ty.display(db).to_string());
         }
 
         item.set_relevance(CompletionRelevance {
             type_match: compute_type_match(completion, &ty),
             exact_name_match: compute_exact_name_match(completion, &name),
-            is_local: true,
+            is_local,
```

Comment:
Should the rest of `CompletionRelevance` be configured here? I assume options like `is_name_already_imported` are important.

Reply:
`is_name_already_imported ` is only used by `use` item completions, so we can ignore this here as its being set somewhere else already (in this case its overwriting the relevance calculated here)

Reply:
Also instead of passing `is_local` as a param to the closre we can put `is_local: matches!(resolution, ScopeDef::Local(_))` here. That simplifies things a bit. (I noticed you also set it to true for `ImplSelfType` but that property is specifically meant for local variables, the doc comment on that isn't clear on that.)

Reply:
We should set the `requires_import` field for it here as well if `import_to_add` (the param of the function we are in right now is `Some`). The rest we can ignore I think.